hoomd/operations.py

```diff
@@ -1,47 +1,112 @@
+    """
+
+    def __init__(self):
         self._compute = list()
         self._scheduled = False
-        self._updaters = SyncedList(OnlyType(_Updater),
+        self._updaters = SyncedList(OnlyType(Updater),
                                     _triggered_op_conversion)
-        self._analyzers = SyncedList(OnlyType(_Analyzer),
+        self._analyzers = SyncedList(OnlyType(Analyzer),
                                      _triggered_op_conversion)
-        self._tuners = SyncedList(OnlyType(_Tuner), lambda x: x._cpp_obj)
+        self._tuners = SyncedList(OnlyType(Tuner), lambda x: x._cpp_obj)
         self._integrator = None
 
         self._tuners.append(ParticleSorter())
 
-    def add(self, op):
+    def add(self, operation):
+        """Add operation to this container.
+
+        Adds the provide operation to the appropriate attribute of the
+        `Operations` instance.
+
+        Args:
+            operation (``operation``): A HOOMD-blue updater, analyzers, compute,
+                tuner, or integrator to add to the collection.
+
+        Raises:
+            RuntimeError: raised when an operation belonging to to this or
+                another `Operations` instance is passed.
+            TypeError: raised when the passed operation is not of a valid type.
+
+        Note:
+            Since only one integrator can be associated with an `Operations`
+            object at a time, this removes the current integrator when called
+            with an integrator operation. Also, the ``integrator`` property
+            cannot be set to ``None`` using this function. Use
+            ``operations.integrator = None`` explicitly for this.
+        """
         # calling _add is handled by the synced lists and integrator property.
         # we raise this error here to provide a more clear error message.
-        if op._added:
+        if operation._added:
             raise RuntimeError(
                 "Operation cannot be added to twice to operation lists.")
-        if isinstance(op, hoomd.integrate._BaseIntegrator):
-            self.integrator = op
+        if isinstance(operation, hoomd.integrate._BaseIntegrator):
+            self.integrator = operation
             return None
-        elif isinstance(op, _Tuner):
-            self._tuners.append(op)
-        elif isinstance(op, _Updater):
-            self._updaters.append(op)
-        elif isinstance(op, _Analyzer):
-            self._analyzers.append(op)
+        elif isinstance(operation, Tuner):
+            self._tuners.append(operation)
+        elif isinstance(operation, Updater):
+            self._updaters.append(operation)
+        elif isinstance(operation, Analyzer):
+            self._analyzers.append(operation)
         else:
-            raise ValueError("Operation is not of the correct type to add to"
-                             " Operations.")
+            raise TypeError(
+                "Operation is not of the correct type to add to Operations.")
```

Comment:
I find that printing invalid types in messages like these helps users debug their code much more efficiently. It helps users reason about what went wrong.
```suggestion
                f"Type {type(operation)} is not a valid type to add to Operations.")
```